Breeding-Insight/bi-docker-stack
# [BI-1998] - Set Up Dev Container For Codespaces

# Description
**Story:** [BI-1998 - Set Up Dev Container For Codespaces](https://breedinginsight.atlassian.net/browse/BI-1998)

Added devcontainer configuration and supporting scripts to enable creation of github codespaces from the bi-docker-stack repository.

Known issues: 
- On first build, the terminal gives a connection aborted/connection reset error. Doing a rebuild consistently fixes this. 
- It takes a few minutes after starting up codespace for DeltaBreed to be fully set up, trying to login prematurely results in an error but after waiting that problem goes away
- Suggestions on how to fix these welcome, otherwise can be a bug card

# Dependencies
n/a

# Testing
- From this branch of bi-docker-stack, create a codespace following instructions in the [Confluence guide] (https://breedinginsight.atlassian.net/wiki/spaces/BI/pages/1951367171/GitHub+Codespaces)
- Check that one can log into the codespace 
- Test that various DeltaBreed functions work (ie importing files, creating new ontology term, etc)
- Delete codespace unless you want to continue playing around with it

# Checklist:
